source/MaterialXCore/Element.cpp

```diff
@@ -326,6 +326,23 @@ ConstElementPtr Element::getRoot() const
     return root;
 }
 
+const string& Element::getActiveColorSpace(bool returnClosest) const
+{
+    ConstElementPtr colorSpaceElement = nullptr;
+    for (ConstElementPtr elem = getSelf(); elem; elem = elem->getParent())
+    {
+        if (elem->hasColorSpace())
+        {
+            colorSpaceElement = elem;
+            if (returnClosest)
```

Comment:
The logic changes is to allow continue traverse to the highest level parent to find possible local targets.